taskfile/reader.go

```diff
@@ -244,10 +244,14 @@ func (r *Reader) readNode(node Node) (*ast.Taskfile, error) {
 				// If there is a cached hash, but it doesn't match the expected hash, prompt the user to continue
 				prompt = fmt.Sprintf(taskfileChangedPrompt, node.Location())
 			}
+
 			if prompt != "" {
+				r.graph.Lock()
```

Comment:
> I've choose to re-use the mutex on the graph.
>
> Maybe you have a better solution @pd93

We probably shouldn't lock all graph writes just for the prompts IMO. For example if the prompt for Taskfile A is accepted, we shouldn't block writing it to the graph because the a prompt for Taskfile B has not been accepted yet. A separate prompt `sync.Mutex` is probably a good idea.

Independently, we might want to consider changing the regular graph mutex to a `sync.RWMutex` at some point. Reads don't need to block one another. I'm not sure why I didn't do this originally.

Reply:
> A separate prompt sync.Mutex is probably a good idea.

:+1: It was my second idea
I've done It

> Independently, we might want to consider changing the regular graph mutex to a sync.RWMutex at some point. Reads don't need to block one another. I'm not sure why I didn't do this originally.

Yeah agree, let's refactor this in another PR :)